conda/connection.py

```diff
@@ -107,8 +108,11 @@ def get_binstar_token(url):
             return None
 
     @staticmethod
-    def is_binstar_url(url):
+    def is_binstar_url_needing_token(url):
         urlparts = urlparse(url)
+        token, path = split_token(urlparts.path)
+        if token:  # url already has token
+            return False
         return (urlparts.scheme in ('http', 'https') and
                 any(urlparts.hostname.endswith(bh) for bh in context.binstar_hosts))
```

Comment:
Why `hostname.endswith` and not an exact (case-insensitive) match?  Are we making an assumption here about the nature/structure of hostnames for the server, and if so, should we explicitly document that?  (e.g. that it's impossible to exclude a sub-domain `foo.bar.com` if `bar.com` is in the list of `binstar_hosts`?)